src/display/Display.spec.js

```diff
@@ -1 +1,46 @@
-// Test away!
+import React from 'react'
+import renderer from 'react-test-renderer'
+import { render, fireEvent } from 'react-testing-library'
+import 'react-testing-library/cleanup-after-each'
+
+import Display from './Display'
+
+describe('<Display />', () => {
+  it('should match snapshot', () => {
+    const tree = renderer.create(<Display />).toJSON()
+    expect(tree).toMatchSnapshot()
+  })
+  it('should render', () => {
+    render(<Display closed="true" locked="true" />)
+  })
+  it('should default to unlocked', () => {
+    const { getByText } = render(<Display />)
+    getByText(/unlocked/i)
+  })
+  it('should default to open', () => {
+    const { getByText } = render(<Display />)
+    getByText(/open/i)
+  })
+  it('should be locked', () => {
+    const { getByText } = render(<Display closed={true} locked={true} />)
+    getByText(/locked/i)
+  })
+  it('should be closed', () => {
+    const { getByText } = render(<Display closed={true} locked={true} />)
+    getByText(/closed/i)
+  })
+  it('should be unlocked', () => {
+    const { getByText } = render(<Display closed={false} locked={false} />)
+    getByText(/unlocked/i)
+  })
+  it('should be green-led', () => {
+    const { getByText } = render(<Display closed={false} locked={false} />)
+    expect(getByText(/open/i).classList[1]).toEqual('green-led')
+    expect(getByText(/unlocked/i).classList[1]).toEqual('green-led')
+  })
+  it('should be red-led', () => {
+    const { getByText } = render(<Display closed={true} locked={true} />)
+    expect(getByText(/closed/i).classList[1]).toEqual('red-led')
+    expect(getByText(/locked/i).classList[1]).toEqual('red-led')
```

Comment:
This test looks identical to the last one minus a few subtleties; do you think you could make a helper function to create these test functions for you? For example:
```javascript
const makeColorTest = ({ closed, locked, color, texts }) => () => {
  const { getByText } = render(<Display { closed, locked } />)
  texts.forEach(text => {
    expect(getByText(text).classList[1]).toEqual(`${color}-led`)
  })
}
```
and its usage for the two tests:
```javascript
it('should be green-led', 
  makeColorTest({
    closed: false,
    locked: false,
    color: 'green'
    texts: [ /open/i, /unlocked/i ]
  })
)

it('should be red-led', 
  makeColorTest({
    closed: true,
    locked: true,
    color: 'red'
    texts: [ /closed/i, /locked/i ]
  })
)
```